example_test.go

```diff
@@ -0,0 +1,338 @@
+	var e3 = Employee{ID: 3, TeamID: 1, Name: "Marco"}
+	var e4 = Employee{ID: 4, TeamID: 2, Name: "Pedro"}
+	var e5 = Employee{ID: 5, TeamID: 3, Name: "Serdar"}
+	var e6 = Employee{ID: 6, TeamID: 3, Name: "Lina"}
+	var e7 = Employee{ID: 7, TeamID: 4, Name: "Joe"}
+	var e8 = Employee{ID: 8, TeamID: 5, Name: "Ben"}
+	var e9 = Employee{ID: 9, TeamID: 5, Name: "Jenny"}
+	var e10 = Employee{ID: 10, TeamID: 6, Name: "Sam"}
+	var e11 = Employee{ID: 11, TeamID: 7, Name: "Melody"}
+	var e12 = Employee{ID: 12, TeamID: 8, Name: "Mark"}
+	var e13 = Employee{ID: 13, TeamID: 8, Name: "Gustavo"}
+	var employees = []Employee{e1, e2, e3, e4, e5, e6, e7, e8, e9, e10, e11, e12, e13}
+	for _, e := range employees {
+		err := db.Query(nil, insertEmployee, e).Run()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Statement to populate the teams table.
+	insertTeam := sqlair.MustPrepare(`
+		INSERT INTO teams (id, team_name, room_id)
+		VALUES ($Team.id, $Team.team_name, $Team.room_id);`,
+		Team{},
+	)
+
+	var t1 = Team{ID: 1, RoomID: 1, Name: "Engineering"}
+	var t2 = Team{ID: 2, RoomID: 1, Name: "Management"}
+	var t3 = Team{ID: 3, RoomID: 1, Name: "Presentation Engineering"}
+	var t4 = Team{ID: 4, RoomID: 2, Name: "Marketing"}
+	var t5 = Team{ID: 5, RoomID: 3, Name: "Legal"}
+	var t6 = Team{ID: 6, RoomID: 3, Name: "HR"}
+	var t7 = Team{ID: 7, RoomID: 4, Name: "Sales"}
+	var t8 = Team{ID: 8, RoomID: 5, Name: "Leadership"}
+	var teams = []Team{t1, t2, t3, t4, t5, t6, t7, t8}
+	for _, t := range teams {
+		err := db.Query(nil, insertTeam, t).Run()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Example 1
+	// Find the team the employee 1 works in.
+	selectSomeoneInTeam := sqlair.MustPrepare(`
+		SELECT &Team.*
+		FROM teams
+		WHERE id = $Employee.team_id`,
+		Employee{}, Team{},
+	)
+
+	// Get returns a single result.
+	var team = Team{}
+	err = db.Query(nil, selectSomeoneInTeam, e1).Get(&team)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("%s is on the %s team.\n", e1.Name, team.Name)
+
+	// Example 2
+	// Find out who is in location l1 and what team they work for.
+	selectPeopleInRoom := sqlair.MustPrepare(`
+		SELECT e.* AS &Employee.*, (t.team_name, t.id) AS (&Team.*)
+		FROM employees AS e, teams AS t
+		WHERE t.room_id = $Location.room_id AND t.id = e.team_id`,
+		Employee{}, Team{}, Location{},
+	)
+
+	// GetAll returns all the results.
+	var roomDwellers = []Employee{}
+	var dwellersTeams = []Team{}
+	err = db.Query(nil, selectPeopleInRoom, l1).GetAll(&roomDwellers, &dwellersTeams)
+	if err != nil {
+		panic(err)
+	}
+
+	for i := range roomDwellers {
+		fmt.Printf("%s (%s), ", roomDwellers[i].Name, dwellersTeams[i].Name)
+	}
+	fmt.Printf("are in %s.\n", l1.Name)
+
+	// Example 3
+	// Cycle through employees until we find one in the Penthouse.
+
+	// A map with a key type of string is used to
+	// pass arguments that are not fields of structs.
+	// sqlair.M is of type map[string]any but if
+	// the map has a key type of string it can be used.
```

Comment:
I think what we want to say in this comment is that you need to create a named type like sqlair.M and that the only requisite is that the keys are strings. Is that the case?

Reply:
Yea that is the case, changed the comment to clarify.